src/status_im/i18n.cljs

```diff
@@ -52,73 +53,181 @@
+                             :sr-RS_#Latn sr-rs-latn/translations
+                             :sr-RS_#Cyrl sr-rs-cyrl/translations
+                             :sv          sv/translations
+                             :sw          sw/translations
+                             :th          th/translations
+                             :tr          tr/translations
+                             :uk          uk/translations
+                             :ur          ur/translations
+                             :vi          vi/translations
+                             :zh          zh-hans/translations
+                             :zh-hans     zh-hans/translations
+                             :zh-hans-tw  zh-hans/translations
+                             :zh-hans-sg  zh-hans/translations
+                             :zh-hans-hk  zh-hans/translations
+                             :zh-hans-cn  zh-hans/translations
+                             :zh-hans-mo  zh-hans/translations
+                             :zh-hant     zh-hant/translations
+                             :zh-hant-tw  zh-hant/translations
+                             :zh-hant-sg  zh-hant/translations
+                             :zh-hant-hk  zh-hant/translations
+                             :zh-hant-cn  zh-hant/translations
+                             :zh-hant-mo  zh-hant/translations
+                             :zh-wuu      zh-wuu/translations
+                             :zh-yue      zh-yue/translations})
+
+;; english as source of truth
+(def labels (set (keys en/translations)))
+
+(spec/def ::label labels)
+(spec/def ::labels (spec/coll-of ::label :kind set? :into #{}))
+
+(defn labels-for-all-locales []
+  (->> translations-by-locale
+       (mapcat #(-> % val keys))
+       set))
+
+;; checkpoints
+
+;; Checkpoints specify milestones for locales.
+;;
+;; With milestones we can ensure that expected supported languages
+;; are actually supported, and visualize the translation state for
+;; the rest of locales according to these milestones.
+;;
+;; Checkpoints are defined by indicating the labels that need to be present
+;; in a locale to achieve that checkpoint.
+;;
+;; We need to define the checkpoint that needs to be achieved for
+;; a locale to be considered supported. This is why as we develop
+;; we add translations, so we need to be defining a new target
+;; for supported languages to achieve.
+;;
+;; Checkpoints are only used in dev and test. In dev when we want to
+;; manually check the state of checkpoints for locales, and in test
+;; to automatically check supported locales against the target checkpoint.
+
+(spec/def ::checkpoint.id keyword?)
+(spec/def ::checkpoint-defs (spec/map-of ::checkpoint.id ::labels))
+
+(def checkpoint-1-labels labels)
+;; these could be status versions instead of incremental numbers
+(def checkpoints-def (spec/assert ::checkpoint-defs
+                                  {::checkpoint-1 checkpoint-1-labels
```

Comment:
Please clarify this section for semi-technical people with to something like (just sketch, not checked and probably better ways to illustrate):

```
;; We start with this state as source of truth, state as of 0.9.12
(def checkpoint-1-labels labels)

;; NOTE: example checkpoint, not actually used
(def checkpoint-2-label (set/union checkpoint-1-labels #{:foo :bar})

;; these could be status versions instead of incremental numbers
;; NOTE: This defines the scope of each checkpoint. To support a checkpoint,
;; change checkpoint-to-consider-locale-supported a few lines below.
(def checkpoints-def
  (spec/assert ::checkpoint-defs
               {::checkpoint-1 checkpoint-1-labels
                ::checkpoint-2 checkpoint-2-labels}))
(def checkpoints (set (keys checkpoints-def)))
```